src/mapper/pkg/resolvers/helpers.go

```diff
@@ -73,3 +111,70 @@ func (i *intentsHolder) GetIntentsPerService(namespaces []string) map[model.Otte
 	}
 	return result
 }
+
+func (i *intentsHolder) getConfigMapNamespace() (string, error) {
+	namespace := viper.GetString(config.NamespaceKey)
+	if namespace != "" {
+		return namespace, nil
+	}
+	namespace, err := kubeutils.GetCurrentNamespace()
+	if err != nil {
```

Comment:
if that's the case, you'd probably want to fail on initialization of `NewIntentsHolder` (and not much later when you already have some info you're trying to write).

Reply:
now with the config it will be as you expcet